Conditionally add IJulia 1.x kernel only if the user has it installed for themselves

MorganRodgers opened the pull request:
Fixes issue described in Incident INC0345013.

```
I found the current setup is not working if a user has installed packages in home directory.

$ module load julia/1.0.0
$ julia
julia> import IJulia
[ Info: Precompiling IJulia [7073ff75-c697-5162-941a-fcdaad2a7d2a]
(v1.0) pkg> add Optim
[ .. skipped .. ]
julia> import IJulia
ERROR: ArgumentError: Package IJulia not found in current path:
- Run `Pkg.add("IJulia")` to install the IJulia package.

This is because Julia uses registries from the global project ($OSC_JULIA_DIR/site/environments/v1.0/Project.toml) while the home project (~/.julia/environments/v1.0/Project.toml) is empty. I don't find any way to combine both projects.

I think the best way is to let users manage their own IJulia kernel. In bc_osc_jupyter, we can point to $HOME/.julia/packages/IJulia/<version_tag>/src/kernel.jl. The question is how to let users to chose the version tag? 
```

johrstrom commented:
I removed my ~/.julia directory to test this and now I don't see julia 1.0 at all.  Are we not going to use the system's provided version either?

MorganRodgers commented:
I suppose that's a fair question. I have removed the Julia 1.0 kernel, but I could keep it in place and conditionally remove it from the Hash? I've updated the SNow incident to ask ZQ and Summer for their opinions.

johrstrom commented:
If you want to pull this in now, and add that some later point, I'm OK with that.  It's a choice between dropping support for users who don't have (by pulling this in now) or breaking users who do have it (by not pulling it in).
